codec/src/main/java/io/netty/handler/codec/HeadersUtils.java

```diff
@@ -230,8 +230,10 @@ public boolean contains(Object o) {
 
         private void fillArray(Object[] arr) {
```

Comment:
Why is this not passed in as a String[] ?

Reply:
Good point. I guess its because the signature of `public <X> X[] toArray(X[] a)` is generic and it is still shared between both `toArray` methods. However we should be able to avoid forcing the `toString` conversion in the case of the `public Object[] toArray()` method. Also it the creation of a `Object[]` array here may not be correct ... we should consider doing something similar to `AbstractCollection`:

```java
T[] r = a.length >= size ? a :
                  (T[])java.lang.reflect.Array
                  .newInstance(a.getClass().getComponentType(), size);
```

Reply:
@Scottmitch I disagree about not doing the string conversion. I think we must do it even if `Object[]` or `CharSequence[]` is passed in as we define it as `Set<String>` which should give the guarantee that only `String` will be stored in there. I will change how we create the Array tho.